### PR TITLE
[SPARK-5474][Build]curl should support URL redirection in build/mvn

### PR DESCRIPTION
**Pull request by witgo**

### CODE REVIEW

**Comment by shaneknapp** [comment]
i will kick this off again once i restart jenkins.

**Comment by shaneknapp** [comment]
jenkins, test this please.

**Comment by srowen** [comment]
Are these all the usages of curl? Would be good to do this for all of them. Same for whet.

**Comment by witgo** [comment]
I test in Max OS x and centos 6.4 system, there is no problem.

**Comment by pwendell** [comment]
@witgo can you update other cases in the codebase where we do this?

**Comment by nchammas** [comment]
Do we need to do something similar for `wget`? Also, what's the motivation for this change? Not that it's bad or anything; just curious.

File: build/mvn [hunk 1]
```diff
@@ -37,10 +37,10 @@ install_app() {
   local curl_opts=""
```
Comment (nchammas) [inline build/mvn]: Minor suggestion: Maybe change this to `local curl_opts="-L"` and update the block below to append options to this list? That way we don't have to duplicate the option.

**Comment by witgo** [comment]
@pwendell  It seems that only this URL(download maven) might return 3xx.

**Pull request closed without merging**